presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -502,6 +505,39 @@ public PlanNode visitFilter(FilterNode node, RewriteContext<Set<VariableReferenc
             return new FilterNode(node.getSourceLocation(), node.getId(), node.getStatsEquivalentPlanNode(), source, node.getPredicate());
         }
 
+        @Override
+        public PlanNode visitCteConsumer(CteConsumerNode node, RewriteContext<Set<VariableReferenceExpression>> context)
+        {
+            // Some output can be pruned but current implementation of PhysicalCteProducer does not allow cteconsumer pruning
+            return node;
+        }
+
+        @Override
+        public PlanNode visitCteProducer(CteProducerNode node, RewriteContext<Set<VariableReferenceExpression>> context)
+        {
+            Set<VariableReferenceExpression> expectedInputs = new HashSet<>();
+            expectedInputs.addAll(node.getOutputVariables());
```

Comment:
Can be combined into one line `Set<VariableReferenceExpression> expectedInputs = ImmutableSet.copyOf(node.getOutputVariables());`